src/solvers/nonlinear_solid_solver.cpp

```diff
@@ -187,13 +190,13 @@ void NonlinearSolidSolver::completeSetup()
 
   // Set the MFEM abstract operators for use with the internal MFEM solvers
   if (timestepper_ == serac::TimestepMethod::QuasiStatic) {
-    nonlinear_oper_                 = std::make_shared<NonlinearSolidQuasiStaticOperator>(std::move(H_form));
     solver_.solver().iterative_mode = true;
+    nonlinear_oper_                 = std::make_unique<NonlinearSolidQuasiStaticOperator>(std::move(H_form));
     solver_.SetOperator(*nonlinear_oper_);
   } else {
-    timedep_oper_ = std::make_shared<NonlinearSolidDynamicOperator>(
-        std::move(H_form), std::move(S_form), std::move(M_form), ess_bdr_, solver_.solver(), lin_params_);
     solver_.solver().iterative_mode = false;
+    timedep_oper_                   = std::make_unique<NonlinearSolidDynamicOperator>(
+        std::move(H_form), std::move(S_form), std::move(M_form), ess_bdr_, solver_.solver(), lin_params_);
```

Comment:
Following the recent discussion about satisfying finite-deformation constraints, if we're going to implement our own nonlinear solvers, will we still need these separate operators? @jamiebramwell

Reply:
Do you mean the time dependent operator vs the quasi-static one? I think it's still useful to have these algebraic constructs as separate objects since they are doing distinct algebraic solves (granted they are using the same building blocks to construct these solves). The time dependent operator framework is required to use MFEM timesteppers which is definitely useful. The built-in Newton solver, not so much. Maybe we could add the quasi-static functionality to the `TimeDependentOperator`? I welcome any ideas!

Anyway, I think this is an idea worth pursuing, but it probably goes beyond the scope of this PR.

Reply:
You're right, it's not intended to really be a request for this PR! I was just wondering how mfem's time integrators interact with the nonlinear solves. It seems like a time integrator built on the implementations of `Mult()` and `ImplicitSolve()` will encounter the same issues with element inversion that Jerome mentioned.

Reply:
That's a fair point. I think it's worth investigating if we can put use those linearization fixes in with the abstract time integrator framework because it would be nice if we could use sundials. However, if we can't we can't.